pkg/aggregator/demultiplexer.go

```diff
@@ -637,6 +637,7 @@ func sendIterableSeries(serializer serializer.MetricSerializer, start time.Time,
 	err := serializer.SendIterableSeries(series)
 	// if err == nil, SenderStopped was called and it is safe to read the number of series.
 	count := series.SeriesCount()
+	series.IterationStopped()
```

Comment:
I might be missing something, but according to a comment on `IterationStopped`, it should be called by a receiver, but this commit moves it to the sender side. Did the purpose of `IterationStopped` changed?

Reply:
This code is called inside a new goroutine and `serializer.SendIterableSeries` is the receiver (`SendIterableSeries` serializes the series and so iterates over the series).
I wrote a code to automatically calls  `IterationStopped` and `SenderStopped` but didn't open a PR. I will revisite this code when working on `FlushSketches` as it can make the code easier to understand.

Reply:
> serializer.SendIterableSeries is the receiver

Would it make sense to be responsible for calling `IterationStopped` then? The goroutine may be receiver side, the overall context (the file and top-level function) is still on the sender side.